### PR TITLE
Add validation for component.yaml v1.1 configuration file

### PR DESCRIPTION
**Pull request by chira99**
## Purpose
The new `component.yaml v1.1` needs to be validated during `source configuration file validation` step in a component build. So we need to $subject

Resolves: https://github.com/wso2-enterprise/choreo/issues/31962
## Goal
- Proper validation of the updated `dependencies` section in `component.yaml v1.1` file.

## Approach
- Added validation logic for the new dependency type (`connectionReferences`).
- Updated Regex checks for each service type (service, thirdparty, database) to ensure strict validation in both v1.0 and v1.1 component.yaml files
- Added tests to cover the validation of v1.1 and updated tests of v1.0 to cover strict validation of each resourceRef field.

### CODE REVIEW

File: test/component-yaml-samples.js [hunk 1]
```diff
@@ -216,51 +216,63 @@ endpoints:
 const validateServiceReferenceName = `schemaVersion: 1.0
 dependencies:
     serviceReferences:
+    #0
```
Comment (binoyPeries) [inline test/component-yaml-samples.js]:
```suggestion
 
```
Please remove these numbers

**Review by binoyPeries — CHANGES_REQUESTED** [review]

**Comment by yashodgayashan** [comment]
@chira99 please note that you need to update the CICD component as well.

File: test/component-yaml-samples.js (hunk 1 above)
Comment (chira99) [reply test/component-yaml-samples.js]: Resolved with e429eddf8435941578956923d56fe5b1f8ada691

**Comment by chira99** [comment]
> @chira99 please note that you need to update the CICD component as well.

ACK

**Review by binoyPeries — APPROVED** [review]

**Comment by yashodgayashan** [comment]
@chira99 do we have an issue to track this? If not please create one and complete it because we are doing Argo parallel and this can cause a BREAKING CHANGE.

**Pull request merged**